website/src/pages/documents/document.page.tsx

```diff
@@ -128,28 +128,31 @@ const TranslationTab = ({ doc }: { doc: Document }) => {
 
   const dialog = useDialogState({ animated: true })
 
-  const [selectedWord, setSelectedWord] =
-    useState<Dailp.FormFieldsFragment | null>(null)
+  const [selectedSegment, setSelectedSegment] = useState<PanelSegment | null>(
+    null
+  )
+
+  const selectAndShowContent = (content: PanelSegment | null) => {
+    setSelectedSegment(content)
 
-  const selectAndShowWord = (content: Dailp.FormFieldsFragment | null) => {
-    setSelectedWord(content)
     if (content) {
       dialog.show()
     } else {
       dialog.hide()
     }
   }
 
-  // When the mobile version of the word panel is closed, remove any word selection.
+  // When the mobile version of the word panel is closed, remove any segment selection.
   useEffect(() => {
     if (!dialog.visible) {
-      selectAndShowWord(null)
+      selectAndShowContent(null)
     }
   }, [dialog.visible])
 
-  let wordPanelInfo = {
-    currContents: selectedWord,
-    setCurrContents: selectAndShowWord,
+  // This can now be either a word or paragraph.
```

Comment:
Before merging, maybe just take "now" out of this comment since someone looking at the updated code won't see the old code

Reply:
Makes sense!